src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -149,7 +149,7 @@ class CanvasImageLayerRenderer extends CanvasLayerRenderer {
     }
 
     const opacity = layerState.opacity;
-    if (opacity !== canvas.style.opacity) {
+    if (opacity != canvas.style.opacity) {
```

Comment:
`canvas.style.opacity` is a string, so a strict comparison with an explicit `parseInt(canvas.style.opacity)` would be better.

Reply:
in 8dbe8bc34e563729eba4d92ac773287726c24ea9